EIPS/eip-1559.md

```diff
@@ -34,48 +38,174 @@ An important aspect of this fee system is that miners only get to keep the gas p
-* To produce an EIP1559 transactions, the new `GAS_PREMIUM` and `FEECAP` fields are set as follows:
-  * `GAS_PREMIUM` serves as a "tip" to the miner
-  * `FEECAP` serves as the absolute maximum that the transaction sender is willing to pay
-* During transaction execution, for EIP1559 transactions we calculate the cost to the `tx.origin` and the gain to the `block.coinbase` as follows:
-  * Set `GASPRICE = min(BASE_FEE + tx.GasPremium, tx.fee_cap)`
-  * Let `GASUSED` be the gas used during the transaction execution/state transition
-  * The `tx.origin` initially pays `GASPRICE * tx.gasLimit`, and gets refunded `GASPRICE * (tx.gasLimit - GASUSED)`
-  * The `block.coinbase` gains `(GASPRICE - BASE_FEE) * GASUSED`.
-    * If `GASPRICE < BASE_FEE` (due to the `FEECAP`), this means that the `block.coinbase` _loses_ funds from this operation; in this case, we check that the post-balance is non-negative and throw an exception if it is negative.
+*Note: `//` in Python is integer division with rounding toward negative infinity.*
+```python
+from typing import Union, Sequence
+from dataclasses import dataclass, field
+from abc import ABC, abstractmethod
+
+@dataclass
+class TransactionLegacy:
+	account_nonce: int = 0
+	gas_price: int = 0
+	gas_limit: int = 0
+	amount: int = 0
+	payload: bytes = bytes()
+	v: int = 0
+	r: int = 0
+	s: int = 0
+
+@dataclass
+class Transaction1559:
+	account_nonce: int = 0
+	gas_limit: int = 0
+	amount: int = 0
+	payload: bytes = bytes()
+	gas_premium: int = 0
+	fee_cap: int = 0
+	v: int = 0
+	r: int = 0
+	s: int = 0
+
+Transaction = Union[TransactionLegacy, Transaction1559]
+
+@dataclass
+class Block:
+	parent_hash: int = 0
+	uncle_hashes: Sequence[int] = field(default_factory=list)
+	author: int = 0
+	state_root: int = 0
+	transaction_root: int = 0
+	transaction_receipt_root: int = 0
+	logs_bloom: int = 0
+	difficulty: int = 0
+	number: int = 0
+	gas_target: int = 0 # note the name change to gas_target from gas_limit
+	gas_used: int = 0
+	timestamp: int = 0
+	extra_data: bytes = bytes()
+	proof_of_work: int = 0
+	nonce: int = 0
+	base_fee: int = 0 # default to 0 for blocks before INITIAL_FORK_BLOCK_NUMBER
+
+@dataclass
+class Account:
+	nonce: int = 0
+	balance: int = 0
+	storage_root: int = 0
+	code_hash: int = 0
+
+INITIAL_FORK_BLOCK_NUMBER = 10 # TBD
+MIGRATION_DURATION_IN_BLOCKS = 800000
+FINAL_FORK_BLOCK_NUMBER = INITIAL_FORK_BLOCK_NUMBER + MIGRATION_DURATION_IN_BLOCKS
+BASE_FEE_MAX_CHANGE_DENOMINATOR = 8
+
+class World(ABC):
+	def validate_block(self, block: Block) -> None:
+		parent_base_fee = self.parent(block).base_fee
+		parent_gas_used = self.parent(block).gas_used
+		parent_gas_target = self.parent(block).gas_target
+		transactions_all = self.transactions(block)
+
+		# check if the block changed the gas target too much
+		assert block.gas_target <= parent_gas_target + parent_gas_target // 1024, 'invalid block: gas target increased too much'
```

Comment:
Hi there, we've started with the EIP-1559 [TypeScript implementation](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1148), will drop some comments here and there if I stumble upon something in the spec (not sure where to do best TBH, does this make sense (as I've done now) to search for the old PRs with the respective changes?).

So first thing here: where is this `1024` coming from? Should this eventually also get a name and a dedicated constant in the section above (`INITIAL_FORK_BLOCK_NUMBER`,...)?

Reply:
It is from before EIP-1559, nothing changed.  It is included here simply because we are touching code related to gas limits and wanted to make it clear that this isn't changing behavior.

Reply:
It is better to ask in the Ethereum R&D Discord server in #eip-1559-dev channel or in the `discussions-to` link on this EIP.

Reply:
Ok, thanks, I will do that.